samples/extensions/raytracing_basic/raytracing_basic.cpp

```diff
@@ -238,266 +136,377 @@ uint64_t RaytracingBasic::get_buffer_device_address(VkBuffer buffer)
+	acceleration_structure_build_range_info.transformOffset                                          = 0;
+	std::vector<VkAccelerationStructureBuildRangeInfoKHR *> acceleration_build_structure_range_infos = {&acceleration_structure_build_range_info};
+
+	// Accelerations structures can be build either on the host or the device, with the former only possible if the implementation supports it
+	if (acceleration_structure_features.accelerationStructureHostCommands)
+	{
+		// Implementation supports building acceleration structure building on host
+		vkBuildAccelerationStructuresKHR(
+		    device->get_handle(),
+		    VK_NULL_HANDLE,
+		    1,
+		    &acceleration_build_geometry_info,
+		    acceleration_build_structure_range_infos.data());
+	}
+	else
 	{
-		memcpy(data, shader_handle_storage.data() + i * ray_tracing_properties.shaderGroupHandleSize, ray_tracing_properties.shaderGroupHandleSize);
-		data += ray_tracing_properties.shaderGroupBaseAlignment;
+		// Build on the device, which involves a command buffer submission
+		VkCommandBuffer command_buffer = get_device().create_command_buffer(VK_COMMAND_BUFFER_LEVEL_PRIMARY, true);
+		vkCmdBuildAccelerationStructuresKHR(
+		    command_buffer,
+		    1,
+		    &acceleration_build_geometry_info,
+		    acceleration_build_structure_range_infos.data());
+		get_device().flush_command_buffer(command_buffer, queue);
 	}
-	shader_binding_table->unmap();
+
+	delete_scratch_buffer(scratch_buffer);
+
+	// Get the top acceleration structure's handle, which will be used to setup it's descriptor
+	VkAccelerationStructureDeviceAddressInfoKHR acceleration_device_address_info{};
+	acceleration_device_address_info.sType                 = VK_STRUCTURE_TYPE_ACCELERATION_STRUCTURE_DEVICE_ADDRESS_INFO_KHR;
+	acceleration_device_address_info.accelerationStructure = top_level_acceleration_structure.handle;
+	top_level_acceleration_structure.device_address        = vkGetAccelerationStructureDeviceAddressKHR(device->get_handle(), &acceleration_device_address_info);
+}
+
+/*
+	Create scene geometry and ray tracing acceleration structures
+*/
+void RaytracingBasic::create_scene()
+{
+	create_bottom_level_acceleration_structure();
+	create_top_level_acceleration_structure();
+}
+
+/*
+	Create the Shader Binding Tables that connects the ray tracing pipelines' programs and the  top-level acceleration structure
+
+	SBT Layout used in this sample:
+
+		/-----------\
+		| raygen    |
+		|-----------|
+		| miss      |
+		|-----------|
+		| hit       |
+		\-----------/
+*/
+
+void RaytracingBasic::create_shader_binding_tables()
+{
+	const uint32_t           handle_size             = ray_tracing_pipeline_properties.shaderGroupHandleSize;
+	const uint32_t           handle_alignment        = ray_tracing_pipeline_properties.shaderGroupHandleAlignment;
+	const uint32_t           group_count             = static_cast<uint32_t>(shader_groups.size());
+	const uint32_t           sbt_size                = handle_alignment * group_count;
```

Comment:
I think this should be handle_size * group_count;

Reply:
I'm not sure on this. Wouldn't this result in the buffer being too small on implementations where `shaderGroupHandleSize` is smaller than `shaderGroupHandleAlignment`? NVidia uses a function to align up for such cases, which I guess would be required here too.

Reply:
Thank you for the update, this is passing for us now.